cmd/promtool/unittest.go

```diff
@@ -326,38 +313,20 @@ Outer:
 			continue
 		}
 
-		var gotSamples []parsedSample
+		var gotSamples samples
 		for _, s := range got {
-			gotSamples = append(gotSamples, parsedSample{
+			gotSamples = append(gotSamples, sample{
 				Labels: s.Metric.Copy(),
 				Value:  s.V,
 			})
 		}
+		sort.Sort(gotSamples)
 
-		var expSamples []parsedSample
-		for _, s := range testCase.ExpSamples {
-			lb, err := promql.ParseMetric(s.Labels)
-			if err != nil {
-				err = errors.Wrapf(err, "labels %q", s.Labels)
-				errs = append(errs, errors.Errorf("    expr: %q, time: %s, err: %s", testCase.Expr,
-					testCase.EvalTime.String(), err.Error()))
-				continue Outer
-			}
-			expSamples = append(expSamples, parsedSample{
-				Labels: lb,
-				Value:  s.Value,
-			})
-		}
-
-		sort.Slice(expSamples, func(i, j int) bool {
-			return labels.Compare(expSamples[i].Labels, expSamples[j].Labels) <= 0
-		})
-		sort.Slice(gotSamples, func(i, j int) bool {
-			return labels.Compare(gotSamples[i].Labels, gotSamples[j].Labels) <= 0
-		})
+		expSamples := testCase.ExpSamples
+		sort.Sort(expSamples)
 		if !reflect.DeepEqual(expSamples, gotSamples) {
-			errs = append(errs, errors.Errorf("    expr: %q, time: %s,\n        exp:%#v\n        got:%#v", testCase.Expr,
-				testCase.EvalTime.String(), parsedSamplesString(expSamples), parsedSamplesString(gotSamples)))
+			errs = append(errs, errors.Errorf("    expr: %q, time: %s\n        exp:%#v\n        got:%#v", testCase.Expr,
```

Comment:
In case of fail the order of "exp_samples" elements can be different then configured by user. Will this not surprise user?